pkg/cli/longhorn.go

```
@@ -0,0 +1,301 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	lhv1b1 "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta1"
+)
+
+var (
+	scaleDownReplicasWaitTime = time.Minute
+)
+
+const (
+	volumeReplicasAnnotation = "kurl.sh/volume-replica-count"
+	longhornNamespace        = "longhorn-system"
+	overProvisioningSetting  = "storage-over-provisioning-percentage"
+)
+
+func NewLonghornCmd(cli CLI) *cobra.Command {
+	return &cobra.Command{
+		Use:   "longhorn",
+		Short: "Perform operations on a longhorn installation within a kURL cluster",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			return cli.GetViper().BindPFlags(cmd.PersistentFlags())
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return cli.GetViper().BindPFlags(cmd.Flags())
+		},
+	}
+}
+
+func NewLonghornRollbackMigrationReplicas(_ CLI) *cobra.Command {
+	return &cobra.Command{
+		Use:          "rollback-migration-replicas",
+		Short:        "Rollback Longhorn volume replicas to their original value.",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			log.Print("Rolling back Longhorn volume replicas to their original value.")
+			cli, err := client.New(config.GetConfigOrDie(), client.Options{})
+			if err != nil {
+				return fmt.Errorf("error creating client: %s", err)
+			}
+			lhv1b1.AddToScheme(cli.Scheme())
+
+			var l1b1Volumes lhv1b1.VolumeList
+			if err := cli.List(cmd.Context(), &l1b1Volumes, client.InNamespace(longhornNamespace)); err != nil {
+				log.Fatalf("error listing longhorn volumes: %s", err)
```

can we return this error rather than log fatal as it is in the other command?